### PR TITLE
Add audit_log forward endpoint

### PR DESCRIPTION
**Pull request by stonezdj**
add config item to set log forward endpoint
  check log endpoint status and cache it, if not active, log info into default log

Signed-off-by: stonezdj <stonezdj@gmail.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/controller/config/controller.go [hunk 1]
```diff
@@ -92,6 +93,22 @@ func (c *controller) UpdateUserConfigs(ctx context.Context, conf map[string]inte
...
+
+func (c *controller) updateLogEndpoint(ctx context.Context, cfgs map[string]interface{}) error {
+	// check if the audit log forward endpoint updated
+	if _, ok := cfgs[common.AuditLogForwardEndpoint]; ok {
+		auditEP := config.AuditLogForwardEndpoint(ctx)
+		if len(auditEP) == 0 {
+			return nil
+		}
+		if active := audit.CheckEndpointActive(auditEP); !active {
+			return errors.BadRequestError(fmt.Errorf("could not connect to the audit endpoint: %v", auditEP))
```
Comment (wy65701436) [inline src/controller/config/controller.go:108]: to avoid SSRF, we should not expose too much message.

File: src/lib/log/audit/audit.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	currentEndpoint string
+	cache           cache.Cache
+	auditLogger     *log.Logger
+}
+
+// Init redirect the audit log to the forward endpoint
+func (a *LoggerManager) Init(ctx context.Context, logEndpoint string) {
+	a.cache = cache.Default()
+	a.currentEndpoint = logEndpoint
+	al, err := syslog.Dial("tcp", logEndpoint,
```
Comment (wy65701436) [inline src/lib/log/audit/audit.go:48]: defer al.Close()？

File: src/lib/log/audit/audit.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	}
+	return log.DefaultLogger()
+}
+
+// IsActive fetch the endpoint status from cache
+func (a *LoggerManager) IsActive(ctx context.Context, address string) bool {
+	var val string
+	cacheKey := fmt.Sprintf("endpoint:%v", address)
+	err := a.cache.Fetch(ctx, cacheKey, &val)
+	if err != nil {
```
Comment (chlins) [inline src/lib/log/audit/audit.go:76]: do we need print error?

File: src/lib/log/audit/audit.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		return
+	}
+	if a.currentEndpoint == address {
+		a.SetEndpointStatus(ctx, address, CheckEndpointActive(address))
+		return
+	}
+
+	// If there are multiple harbor-core instances, one of the harbor-core change the endpoint config
+	// other harbor-core instances should switch audit log to the new endpoint
+	log.Info("Refresh audit log forward endpoint")
```
Comment (chlins) [inline src/lib/log/audit/audit.go:95]: confuse here, different harbor core instances override one key?

File: src/lib/log/audit/audit.go [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	"github.com/goharbor/harbor/src/lib/config"
+	"github.com/goharbor/harbor/src/lib/log"
+)
+
+const (
+	endpointHealthCheckInterval = 1 * time.Minute
+	cacheInterval               = 2 * time.Minute
+)
+
+var LogMgr = &LoggerManager{}
```
Comment (wy65701436) [inline src/lib/log/audit/audit.go:35]: why do we put the audit log manager into lib?

File: src/lib/log/audit/audit.go [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+	if err != nil {
+		log.Errorf("failed to cache endpoint status, error %v, the audit log may log to standard log", err)
+	}
+	a.currentEndpoint = address
+}
+
+// StartEndpointHealthCheck refresh the endpoint status periodically
+func StartEndpointHealthCheck(ctx context.Context, closing, done chan struct{}) {
+	auditEP := config.AuditLogForwardEndpoint(ctx)
+	LogMgr.Init(ctx, auditEP)
```
Comment (wy65701436) [inline src/lib/log/audit/audit.go:130]: the init here is in a gorountine, there is no guarantee that init will be executed successfully before it is used.

File: src/lib/log/audit/audit.go [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+		a.SetEndpointStatus(ctx, logEndpoint, false)
+		return
+	}
+	a.auditLogger = log.New(al, log.NewTextFormatter(), log.InfoLevel, 3)
+	a.SetEndpointStatus(ctx, logEndpoint, true)
+}
+
+// DefaultLogger ...
+func (a *LoggerManager) DefaultLogger(ctx context.Context) *log.Logger {
+	endpoint := config.AuditLogForwardEndpoint(ctx)
```
Comment (wy65701436) [inline src/lib/log/audit/audit.go:61]: can we just use it and not update any values? If the server can't connect, let it fail.

File: src/lib/log/audit/audit.go (hunk 5 above)
Comment (stonezdj) [reply src/lib/log/audit/audit.go:26]: move to pkg/audit

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**